sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/workflow/WorkflowImpl.scala

```diff
@@ -207,7 +207,7 @@ final class WorkflowImpl(system: ActorSystem, val services: Map[String, Workflow
         val protoEffect =
           persistence match {
             case UpdateState(newState) =>
-              router._internalSetInitState(newState)
+              router._internalSetInitState(newState, transition.isInstanceOf[End.type])
```

Comment:
Maybe it will be better to fail if the user returns an Effect with persistence set to `UpdateState`. 

I would say, if finished, only `NoPersistence` and `DeleteState` (not yet supported), should be allowed.

Reply:
The same applies to the Transition. Once finished, all Effects should have a `NoTransition` and fail otherwise. 

As such we close the doors and only allow methods that are effectively read-only.

Reply:
The thing is that if I throw an exception here then the proxy will just return a generic "unexpected error" to the user. We can log a warning here (or info), but I would create a separate issue to focus on the response on the proxy side.

Reply:
Can't we introduce a specific failure for this so that the proxy is then able to convert to something more sensible?

Reply:
That would require an additional protocol change. I think that the proxy has already all the information to construct a detailed response.

Reply:
And the logic to validate the effect must be present in the proxy anyway, since in theory, anyone can create an SDK.